src/services/wallets/EthWalletConnector.ts

```diff
@@ -0,0 +1,127 @@
+import { AoSigner, ContractSigner } from '@ar.io/sdk/web';
+import { MetamaskError } from '@src/utils/errors';
+import { InjectedEthereumSigner, createData } from 'arbundles';
+import { PermissionType } from 'arconnect';
+import { ApiConfig } from 'arweave/node/lib/api';
+import { hashMessage, recoverPublicKey, toBytes } from 'viem';
+import {
+  Config,
+  Connector,
+  UseAccountReturnType,
+  UseSignMessageReturnType,
+} from 'wagmi';
+import { DisconnectMutateAsync } from 'wagmi/query';
+
+import {
+  AoAddress,
+  ArNSWalletConnector,
+  EthAddress,
+  WALLET_TYPES,
+} from '../../types';
+
+export const ARCONNECT_WALLET_PERMISSIONS: PermissionType[] = [
+  'ACCESS_ADDRESS',
+  'ACCESS_ALL_ADDRESSES',
+  'ACCESS_PUBLIC_KEY',
+  'SIGN_TRANSACTION',
+  'ACCESS_ARWEAVE_CONFIG',
+];
+
+export class EthWalletConnector implements ArNSWalletConnector {
+  ethAccount: UseAccountReturnType<Config>;
+  contractSigner?: ContractSigner;
+  address: EthAddress;
+  disconnectCallback: DisconnectMutateAsync<unknown>;
+  connector: Connector;
+
+  constructor(
+    ethAccount: UseAccountReturnType<Config>,
+    address: EthAddress,
+    disconnectAsync: DisconnectMutateAsync<unknown>,
+    signMessage: UseSignMessageReturnType<unknown>,
+    connector: Connector,
```

Comment:
recommend moving to a configuration where you pass in the wagmi connector (the one defined in main.tsx) and use methods on that for signing and creating signers.

For reference: https://github.com/project-kardeshev/ao-wallet-kit/blob/2903b5099d10390138767e8b4f5b5ed7e9ad5e15/src/strategy/strategies/Wagmi.ts#L27

Reply:
Done. May have some further opportunities to refine but I think it's in good enough shape now.